client/core/bookie.go

```diff
@@ -213,7 +213,8 @@ func (b *bookie) logEpochReport(note *msgjson.EpochReportNote) error {
 			Payload: CandleUpdate{
 				Dur:          durStr,
 				DurMilliSecs: uint64(dur.Milliseconds()),
-				Candle:       c,
+				// Have to copy msgjson.Candle data here since it will be used concurrently.
+				Candle: &*c,
```

Comment:
`SA4001: &*x will be simplified to x. It will not copy x. (staticcheck)`

Sorry, I made my suggestion assuming you had change the `CandleUpdate` definition to have a value.

Let's just make the copy a few lines up, before `b.send`